src/aind_ephys_ibl_gui_conversion/ephys.py

```diff
@@ -58,137 +62,355 @@ def extract_spikes(sorting_folder,results_folder):
-            waveform = templates[unit_idx,:,:]
-            peak_channel = np.argmax(np.max(waveform, 0) - np.min(waveform,0))
-            peak_waveform = waveform[:,peak_channel]
-            peakToTrough = (np.argmax(peak_waveform) - np.argmin(peak_waveform)) / 30000.
-            cluster_channels.append(peak_channel)#int(channel_locs[peak_channel,1] / 10))
-            cluster_peakToTrough.append(peakToTrough)
-            cluster_waveforms.append(waveform)
-
-        np.save(phy_folder / "cluster_peakToTrough.npy", np.array(cluster_peakToTrough))
-        np.save(phy_folder / "cluster_waveforms.npy", np.stack(cluster_waveforms))
-        np.save(phy_folder / "cluster_channels.npy", np.array(cluster_channels))
-
-        # rename files
-        _FILE_RENAMES = [  # file_in, file_out
-                ('channel_positions.npy', 'channels.localCoordinates.npy'),
-                ('channel_inds.npy', 'channels.rawInd.npy'),
-                ('cluster_peakToTrough.npy', 'clusters.peakToTrough.npy'),
-                ('cluster_channels.npy', 'clusters.channels.npy'),
-                ('cluster_waveforms.npy', 'clusters.waveforms.npy'),
-                ('spike_clusters.npy', 'spikes.clusters.npy'),
-                ('amplitudes.npy', 'spikes.amps.npy'),
-                ('spike_depths.npy', 'spikes.depths.npy'),
-                ('spike_times.npy', 'spikes.times.npy'),
-            ]
+
+        templates = []
+        channel_locs = []
+        quality_metrics = []
+
+
+        for analyzer in analyzer_mappings:
+            export_to_phy(analyzer, 
+                        output_folder=phy_folder,
+                        compute_pc_features=False,
+                        remove_if_exists=True,
+                        copy_binary=False,
+                        dtype = 'int16')
+
+            spike_locations = analyzer.get_extension("spike_locations").get_data()
+            template_ext = analyzer.get_extension("templates")
+            templates = template_ext.get_templates()
+
+            for unit_idx, unit_id in enumerate(analyzer.unit_ids):
+                waveform = templates[unit_idx,:,:]
+                peak_channel = np.argmax(np.max(waveform, 0) - np.min(waveform,0))
+                peak_waveform = waveform[:,peak_channel]
+                peakToTrough = (np.argmax(peak_waveform) - np.argmin(peak_waveform)) / 30000.
+                cluster_channels.append(peak_channel)#int(channel_locs[peak_channel,1] / 10))
+                cluster_peak_to_trough.append(peakToTrough)
+                cluster_waveforms.append(waveform)
+
+            print('Converting data...')
+
+            if len(clusters) == 0:
+                clusters.append(np.load(phy_folder / "spike_clusters.npy"))
+            else:
+                shank_clusters = np.load(phy_folder / 'spike_clusters.npy')
+                shank_clusters = np.array([cluster + clusters[-1][-1] for cluster in shank_clusters])
+                clusters.append(shank_clusters)
```

Comment:
I think this is assuming the last cluster is the largest one, but this is not the case.
We should do something like this:

We define a `cluster_offset = 0` prior to the loop.
```suggestion
            current_clusters = np.load(phy_folder / "spike_clusters.npy")
            current_clusters = current_clusters + cluster_offset
            cluster_offset =  np.max(current_clusters) + 1
            clusters.append(current_clusters)
```
Does it make sense?

Reply:
yep, this makes a lot of sense!